frontend/src/routes/_protected/treecluster/index.tsx

```diff
@@ -1,13 +1,45 @@
+import { useState, useEffect } from 'react';
 import TreeclusterCard from "@/components/general/cards/TreeclusterCard";
+import Dialog from "@/components/general/filter/Dialog";
 import { treeclusterDemoData } from "@/data/treecluser";
-import { createFileRoute } from "@tanstack/react-router";
+import { createFileRoute, useLoaderData } from '@tanstack/react-router';
+import { z } from 'zod';
 
-export const Route = createFileRoute("/_protected/treecluster/")({
+const treeclusterFilterSchema = z.object({
+  status: z.array(z.string()).optional(),
+  region: z.array(z.string()).optional(),
+});
+
+export const Route = createFileRoute('/_protected/treecluster/')({
   component: Treecluster,
+  validateSearch: treeclusterFilterSchema,
+
+  loaderDeps: ({ search: { status, region } }) => ({
+    status: status || [],
+    region: region || [],
+  }),
```

Comment:
So now we can say that if it's not undefined, take the search param as an array, otherwise use an empty array.